models/modeltrawler.py

```diff
@@ -0,0 +1,117 @@
+import time
+import itertools
+import re
+
+from multiprocessing.pool import ThreadPool
+
+
+def print_progress_decorator(func):
+
+    def func_wrapper(*args, **kwargs):
+        _self = args[0]
+        print_counts = kwargs.pop('print_counts', None)
+
+        for model in func(*args, **kwargs):
+            if print_counts:
+                func_wrapper.counter += 1
+                if func_wrapper.counter % print_counts == 0:
+                    print('{} {} returned in {}s'.format(func_wrapper.counter, _self.model, _self.get_time_running()))
```

Comment:
I'd prefer if we used `logger` methods instead of `print`. It adds the current timestamp to all messages and is generally easier to control verbosity etc.

There are examples of how to do this in the index services and buyer notification scripts.